src/app/public/modules/href/href.directive.ts

```diff
@@ -0,0 +1,222 @@
+import { HttpParams } from '@angular/common/http';
+import {
+  Directive,
+  ElementRef,
+  HostListener,
+  Input,
+  Optional,
+  Renderer2
+} from '@angular/core';
+import { Router, UrlTree } from '@angular/router';
+import { SkyAppConfig, SkyAppRuntimeConfigParamsProvider } from '@skyux/config';
+import { Subject } from 'rxjs';
+
+import { SkyHrefQueryParams } from './href-query-params';
+import { SkyHrefResolverService } from './href-resolver.service';
+
+import { AppRoute } from './types/app-route';
+
+type HrefChanges = { href: string, styleDisplay: string };
+
+@Directive({
+  selector: '[skyHref]'
+})
+export class SkyHrefDirective {
+
+  @Input()
+  public skipLocationChange = false;
+
+  @Input()
+  public replaceUrl = false;
+
+  @Input()
+  public state?: {[k: string]: any};
+
+  @Input()
+  public set queryParams(value: SkyHrefQueryParams) {
```

Comment:
Since the other two `@Input()` properties start with `skyHref`, I'm thinking this one should, too, to avoid any clashes with consumers' directives.

Reply:
@Blackbaud-PaulCrowder these four input parameters match ones from `[routerLink]`. I almost wonder if the selector ought to be `[skyHref]:not([routerLink])` because the two directives should not be combined.